src/services/postServices.ts

```diff
@@ -0,0 +1,92 @@
+import { Document } from "mongoose";
+import { BadRequestError, ServerError } from "../errors/customErrors.ts";
+import PostModel, { Post } from "../models/Post.ts";
+import UserModel from "../models/UserModel.ts";
+
+export const getSinglePostDocument = async (
+  postId: string,
+): Promise<Document> => {
+  try {
+    const post = await PostModel.findById(postId);
+    if (!post) {
+      throw new ServerError("Post not found");
+    }
+    return post.toJSON() as Document;
+  } catch (err) {
+    throw new ServerError(`${err}`);
+  }
+};
+
+export const buildPostDocument = async (body: Post): Promise<string> => {
+  try {
+    const { id, author } = body;
```

Comment:
Why are we having id at here? The post is not created.  The route we send was POST to was not dynamic that having params or query. The id is created after the post is created.